# FINANCIAL DATA ANALYSIS
This analysis focuses on financial statements from top global companies spanning from the year 2009 - 2023.

The goal of this project is to evaluate the Financial health , Performance and strategic outlook of organization among multiply top industries. 
The study of the financial statements will identify trends to financial risk, employment dynamics, investment opportunities,corporate growth stability and inflation rate impact on business performance.
The project specifically focuses on:
1. Financial risk assessment. Profitability and industry-wise revenue distribution and long-term solvency across firms.
2. Employment and workforce patterns. Understanding how hiring, layoffs, labour cost and productivity have shifted overtime.
3. Investment opportunity. Identifying companies and industries with consistence growth indicators and good business modules.
4. Industry comparisons. Examine how different sectors perform during economic cycles.(eg.recovery /growth periods)


In [2]:
# Importing  the necessary libraries

import pandas as pd                    #import pandas for data manipulations
import os                              # Import os module for path operations
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Explanation and Documentation 
In this Section , I am going to read my data file from a local path and a backup source with the if_statement and also to show key financial metrics in my data set(columns).

In [108]:
#Section 1.1. Reading data file
file_path = r"C:\Users\Isaac Oppong Agyei\Downloads\Financial_Statements.csv"
back_up = "Financial_Statements.csv"  

#Check if the file exits in One_Drive/Local path
if os.path.exists(file_path):
    financials = pd.read_csv(file_path)
    print("file loaded sucessfully!")

    #safely reading df if file exist
    print(financials.head())

else:
    financials= pd.read_csv(back_up)
    print(financials.tail())


     Year Company  Category  Market Cap(in B USD)  Revenue  Gross Profit  \
156  2013     AMZN     LOGI                182.54  74452.0       20271.0   
157  2012     AMZN     LOGI                113.63  61093.0       15122.0   
158  2011     AMZN     LOGI                 78.72  48077.0       10789.0   
159  2010     AMZN     LOGI                 80.79  34204.0        7643.0   
160  2009     AMZN     LOGI                 58.24  24509.0        5531.0   

     Net Income  Earning Per Share  EBITDA  Share Holder Equity  ...  \
156       274.0             0.0295  3998.0               9746.0  ...   
157       -39.0            -0.0045  2835.0               8192.0  ...   
158       631.0             0.0685  1945.0               7757.0  ...   
159      1152.0             0.1265  1974.0               6864.0  ...   
160       902.0             0.1020  1507.0               5257.0  ...   

     Current Ratio  Debt/Equity Ratio      ROE     ROA      ROI  \
156         1.0716             0.3274   2.8

In [4]:
# section 1.2. printing data columns and shape:

print(financials.columns)
print("Dataset Rows & Columns:", financials.shape)

Index(['Year', 'Company ', 'Category', 'Market Cap(in B USD)', 'Revenue',
       'Gross Profit', 'Net Income', 'Earning Per Share', 'EBITDA',
       'Share Holder Equity', 'Cash Flow from Operating',
       'Cash Flow from Investing', 'Cash Flow from Financial Activities',
       'Current Ratio', 'Debt/Equity Ratio', 'ROE', 'ROA', 'ROI',
       'Net Profit Margin', 'Free Cash Flow per Share',
       'Return on Tangible Equity', 'Number of Employees',
       'Inflation Rate(in US)'],
      dtype='object')
Dataset Rows & Columns: (161, 23)


#Reflection on results

Importing the necessary libraries that will be needed for the project.
if-statement with the os-path-exists() was use to load the data_set. if the file existed in my One_Drive or local path it load from there otherwise it automatically switched to the backup source to avoid error.
Exploring the data a bit using the tail() and shape, we could see from data shape that our data has 161 rows and 23 columns.
Printing the columns of the data to clean and make it easy to work with we can see the names of all the 23 columns has an object dtype, meaning they store string values.
The data set also contains metrics such as: 
1. Market Cap, Revenue, Gross Profit, Net Income.
2. Financial Ratios: ROE, ROA, Debt/Equity Ratio, Net Profit Margin.
3. Cash Flow Metrics: Operating, Investing, and Financing Activities.
4. Inflation Rate Impact on Business Performance.
5. Number of Employees, Employer Growth, Workforce Stability.
6. EBITDA, Shareholder Equity.

 In this section I will create multiple reusable functions to handle different aspect of our data analysis

In [5]:
#Section 2.1 Explore data with a function 
financials.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 161 entries, 0 to 160
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Year                                 161 non-null    int64  
 1   Company                              161 non-null    object 
 2   Category                             161 non-null    object 
 3   Market Cap(in B USD)                 160 non-null    float64
 4   Revenue                              161 non-null    float64
 5   Gross Profit                         161 non-null    float64
 6   Net Income                           161 non-null    float64
 7   Earning Per Share                    161 non-null    float64
 8   EBITDA                               161 non-null    float64
 9   Share Holder Equity                  161 non-null    float64
 10  Cash Flow from Operating             161 non-null    float64
 11  Cash Flow from Investing        

In [6]:
#section 2.2: cleaning data columns (removing whitespaces, special characters,filling spaces, among others)

def clean_column(col):
    col = col.strip()
    col = col.replace(" ", "_")
    col = col.replace("(", " ")
    col = col.replace(")", " ")
    col = col.upper()
    return col
    
#calling the function by creating a new_column
new_financials_columns = []

for c in financials.columns:
    clean_columns = clean_column(c)
    new_financials_columns.append(clean_columns)

financials.columns = new_financials_columns
print(financials.columns)

Index(['YEAR', 'COMPANY', 'CATEGORY', 'MARKET_CAP IN_B_USD ', 'REVENUE',
       'GROSS_PROFIT', 'NET_INCOME', 'EARNING_PER_SHARE', 'EBITDA',
       'SHARE_HOLDER_EQUITY', 'CASH_FLOW_FROM_OPERATING',
       'CASH_FLOW_FROM_INVESTING', 'CASH_FLOW_FROM_FINANCIAL_ACTIVITIES',
       'CURRENT_RATIO', 'DEBT/EQUITY_RATIO', 'ROE', 'ROA', 'ROI',
       'NET_PROFIT_MARGIN', 'FREE_CASH_FLOW_PER_SHARE',
       'RETURN_ON_TANGIBLE_EQUITY', 'NUMBER_OF_EMPLOYEES',
       'INFLATION_RATE IN_US '],
      dtype='object')


Reflection on results
From reading our data_set columns we could see that the data has a mixture of lower and Uppercase letters, as well as spaces and parenthesis,One noticeable issue is that the 'Company ' column name has a space in front of it. we will write a function to correct all the issues with the column.
The function will be able to clean the columns by; 
 -Replacing spaces between words with underscores.
 -Removing any special characters, like parentheses.
 -Making all labels uppercase.
 -Removing any white spaces.

# Explanation and Documentation 
In this section I am going to use Pandas to handle and clean the data set by handling missing data, removing duplicates,filtering ,aggregating and preparing the data for analysis. This means our financial data will be accurate, clean and ready for insights.

In [7]:
#Section 3.1 Checking and Handling missing values
financials.isnull().sum()

YEAR                                   0
COMPANY                                0
CATEGORY                               0
MARKET_CAP IN_B_USD                    1
REVENUE                                0
GROSS_PROFIT                           0
NET_INCOME                             0
EARNING_PER_SHARE                      0
EBITDA                                 0
SHARE_HOLDER_EQUITY                    0
CASH_FLOW_FROM_OPERATING               0
CASH_FLOW_FROM_INVESTING               0
CASH_FLOW_FROM_FINANCIAL_ACTIVITIES    0
CURRENT_RATIO                          0
DEBT/EQUITY_RATIO                      0
ROE                                    0
ROA                                    0
ROI                                    0
NET_PROFIT_MARGIN                      0
FREE_CASH_FLOW_PER_SHARE               0
RETURN_ON_TANGIBLE_EQUITY              0
NUMBER_OF_EMPLOYEES                    0
INFLATION_RATE IN_US                   0
dtype: int64

In [53]:
#Section 3.2 dropping missing value in row
new_financials= financials.dropna(axis = 0)
new_financials.isnull().sum()

YEAR                                   0
COMPANY                                0
CATEGORY                               0
MARKET_CAP IN_B_USD                    0
REVENUE                                0
GROSS_PROFIT                           0
NET_INCOME                             0
EARNING_PER_SHARE                      0
EBITDA                                 0
SHARE_HOLDER_EQUITY                    0
CASH_FLOW_FROM_OPERATING               0
CASH_FLOW_FROM_INVESTING               0
CASH_FLOW_FROM_FINANCIAL_ACTIVITIES    0
CURRENT_RATIO                          0
DEBT/EQUITY_RATIO                      0
ROE                                    0
ROA                                    0
ROI                                    0
NET_PROFIT_MARGIN                      0
FREE_CASH_FLOW_PER_SHARE               0
RETURN_ON_TANGIBLE_EQUITY              0
NUMBER_OF_EMPLOYEES                    0
INFLATION_RATE IN_US                   0
dtype: int64

In [9]:
#section 3.3 Checking for duplicates in our date_frame
new_financials.duplicated().sum()

0

#section 3.4 Data Filtering
In this section I will filter the data and give certain insight of some of a company, industry category, and performance within certain year.

In [11]:
#Filter data of a specific company.
AMZN_Data = new_financials[new_financials["COMPANY"]== "AMZN"]
AMZN_Data.head()

YEAR COMPANY CATEGORY  MARKET_CAP IN_B_USD    REVENUE  GROSS_PROFIT  \
147  2022    AMZN     LOGI                856.94  513983.0      225152.0   
148  2021    AMZN     LOGI               1691.00  469822.0      197478.0   
149  2020    AMZN     LOGI               1634.16  386064.0      152757.0   
150  2019    AMZN     LOGI                916.15  280522.0      114986.0   
151  2018    AMZN     LOGI                734.42  232887.0       93731.0   

     NET_INCOME  EARNING_PER_SHARE   EBITDA  SHARE_HOLDER_EQUITY  ...  \
147     -2722.0            -0.2700  54169.0             146043.0  ...   
148     33364.0             3.2400  59312.0             138245.0  ...   
149     21331.0             2.0900  48079.0              93404.0  ...   
150     11588.0             1.1505  36330.0              62060.0  ...   
151     10073.0             1.0070  27762.0              43549.0  ...   

     CURRENT_RATIO  DEBT/EQUITY_RATIO      ROE     ROA      ROI  \
147         0.9446             0.4598  -1.8638 -0.5883  -1.2768   
148         1.1358             0.3526  24.1340  7.9334  17.8428   
149         1.0502             0.3406  22.8374  6.6411  17.0348   
150         1.0970             0.3773  18.6723  5.1446  13.5573   
151         1.0981             0.5395  23.1303  6.1931  15.0245   

     NET_PROFIT_MARGIN  FREE_CASH_FLOW_PER_SHARE  RETURN_ON_TANGIBLE_EQUITY  \
147            -0.5296                   -0.2546                    -2.1645   
148             7.1014                   -3.9226                    27.1530   
149             5.5252                    0.4798                    27.2124   
150             4.1309                    0.6220                    24.4958   
151             4.3253                    1.0975                    34.7333   

     NUMBER_OF_EMPLOYEES  INFLATION_RATE IN_US   
147              1541000                 8.0028  
148              1608000                 4.6979  
149              1298000                 1.2336  
150               798000                 1.8122  
151               647500                 2.4426  

[5 rows x 23 columns]

In [13]:
#Filter data Between certain years(eg.Post_covid Era 2020 - 2023)
post_covid = new_financials[(new_financials["YEAR"] >= 2020) & (new_financials["YEAR"] <= 2023)]
post_covid.head()

YEAR COMPANY CATEGORY  MARKET_CAP IN_B_USD    REVENUE  GROSS_PROFIT  \
0   2022    AAPL       IT               2066.94  394328.0      170782.0   
1   2021    AAPL       IT               2913.28  365817.0      152836.0   
2   2020    AAPL       IT               2255.97  274515.0      104956.0   
14  2023    MSFT       IT               2451.23  211915.0      146052.0   
15  2022    MSFT       IT               1787.73  198270.0      135620.0   

    NET_INCOME  EARNING_PER_SHARE    EBITDA  SHARE_HOLDER_EQUITY  ...  \
0      99803.0               6.11  130541.0              50672.0  ...   
1      94680.0               5.61  120233.0              63090.0  ...   
2      57411.0               3.28   77344.0              65339.0  ...   
14     72361.0               9.68  102384.0             206223.0  ...   
15     72738.0               9.65   97843.0             166542.0  ...   

    CURRENT_RATIO  DEBT/EQUITY_RATIO       ROE      ROA      ROI  \
0          0.8794             2.3695  196.9589  28.2924  66.6994   
1          1.0746             1.9768  150.0713  26.9742  54.9839   
2          1.3636             1.7208   87.8664  17.7256  35.0054   
14         1.7692             0.2291   35.0887  17.5644  29.1528   
15         1.7846             0.2989   43.6755  19.9370  34.0575   

    NET_PROFIT_MARGIN  FREE_CASH_FLOW_PER_SHARE  RETURN_ON_TANGIBLE_EQUITY  \
0             25.3096                    1.3146                   196.9589   
1             25.8818                    1.3261                   150.0713   
2             20.9136                    1.0183                    87.8664   
14            34.1462                   -0.6808                    56.1064   
15            36.6863                    1.2643                    82.9207   

    NUMBER_OF_EMPLOYEES  INFLATION_RATE IN_US   
0                164000                 8.0028  
1                154000                 4.6979  
2                147000                 1.2336  
14               221000                 3.7000  
15               221000                 8.0028  

[5 rows x 23 columns]

In [81]:
#Filter data of the companies with High Profit margin
high_profit = new_financials[new_financials["NET_PROFIT_MARGIN"] >20.5]
high_profit.head()

YEAR COMPANY CATEGORY  MARKET_CAP IN_B_USD    REVENUE  GROSS_PROFIT  \
0  2022    AAPL       IT               2066.94  394328.0      170782.0   
1  2021    AAPL       IT               2913.28  365817.0      152836.0   
2  2020    AAPL       IT               2255.97  274515.0      104956.0   
3  2019    AAPL       IT               1304.76  260174.0       98392.0   
4  2018    AAPL       IT                748.54  265595.0      101839.0   

   NET_INCOME  EARNING_PER_SHARE    EBITDA  SHARE_HOLDER_EQUITY  ...  \
0     99803.0               6.11  130541.0              50672.0  ...   
1     94680.0               5.61  120233.0              63090.0  ...   
2     57411.0               3.28   77344.0              65339.0  ...   
3     55256.0               2.97   76477.0              90488.0  ...   
4     59531.0               2.98   81801.0             107147.0  ...   

   CURRENT_RATIO  DEBT/EQUITY_RATIO       ROE      ROA      ROI  \
0         0.8794             2.3695  196.9589  28.2924  66.6994   
1         1.0746             1.9768  150.0713  26.9742  54.9839   
2         1.3636             1.7208   87.8664  17.7256  35.0054   
3         1.5401             1.1940   61.0645  16.3230  30.3113   
4         1.1329             1.0685   55.5601  16.2775  29.6348   

   NET_PROFIT_MARGIN  FREE_CASH_FLOW_PER_SHARE  RETURN_ON_TANGIBLE_EQUITY  \
0            25.3096                    1.3146                   196.9589   
1            25.8818                    1.3261                   150.0713   
2            20.9136                    1.0183                    87.8664   
3            21.2381                   -0.0388                    61.0645   
4            22.4142                    0.7414                    55.5601   

   NUMBER_OF_EMPLOYEES  INFLATION_RATE IN_US   
0               164000                 8.0028  
1               154000                 4.6979  
2               147000                 1.2336  
3               137000                 1.8122  
4               132000                 2.4426  

[5 rows x 23 columns]

Section 3.5 Data Aggregation and Grouping
In this Section I will be doing summary statistics for the data using groupby(),Sum(),mean() which will help us to discover patterns in revenue,profitability,employment and financial performance across industries,companies and years.

In [58]:
# Average Revenue Per Company
Avg_revenue_company = new_financials.groupby("COMPANY")["REVENUE"].mean().round(2).sort_values(ascending=False)
Avg_revenue_company.head()

COMPANY
AAPL    211829.21
AMZN    188247.14
MSFT    111204.33
GOOG    111159.14
INTC     59637.79
Name: REVENUE, dtype: float64

In [ ]:
#Total Market Cap by Category(Industry)
Market_by_Category =financials.groupby("CATEGORY")["MARKET_CAP"].sum().sort_values(ascending=False)
Market_by_Category.head()

In [97]:
#Average employment by Company
Ave_employees = new_financials.groupby("COMPANY")["NUMBER_OF_EMPLOYEES"].mean().round(0).sort_values(ascending=False)
Ave_employees.head()

COMPANY
AMZN     536050.0
MCD      321429.0
SHLDQ    220090.0
MSFT     138267.0
BCS      113486.0
Name: NUMBER_OF_EMPLOYEES, dtype: float64

In [101]:
#Industry Growth in Net Income after post covid_era(2020 - 2023)
Net_income_trend = post_covid.groupby("CATEGORY")["NET_INCOME"].sum().round(2).sort_values(ascending=False)
Net_income_trend.head()

CATEGORY
IT      678819.0
ELEC     70029.0
LOGI     51973.0
FOOD     18453.1
BANK     16938.6
Name: NET_INCOME, dtype: float64

In [104]:
# Companies with Highest Debt to Equity Ratio(Risk Indicator)
high_risk = new_financials[["COMPANY", "DEBT/EQUITY_RATIO"]].sort_values(by = "DEBT/EQUITY_RATIO",ascending=False)
high_risk.head(5)

COMPANY  DEBT/EQUITY_RATIO
116     BCS             9.3328
117     BCS             8.5152
114     BCS             7.6880
115     BCS             7.6045
104     BCS             7.1602

In [106]:
#Key Insights from the data_set after Filtering and Aggregation
print("Key insight extracted:\n")

print("\n1. Highest profitable company")
print(Avg_revenue_company.head(1))

print("\n2. Company with the highest employment")
print(Ave_employees.head(1))

print("\n3. Fast growing industry after post_covid Era(2020 - 2023)")
print(Net_income_trend.head(1))

print("\n4. Company with the highest Debt risk")
print(high_risk.head(1))

Key insight extracted:


1. Highest profitable company
COMPANY
AAPL    211829.21
Name: REVENUE, dtype: float64

2. Company with the highest employment
COMPANY
AMZN    536050.0
Name: NUMBER_OF_EMPLOYEES, dtype: float64

3. Fast growing industry after post_covid Era(2020 - 2023)
CATEGORY
IT    678819.0
Name: NET_INCOME, dtype: float64

4. Company with the highest Debt risk
    COMPANY  DEBT/EQUITY_RATIO
116     BCS             9.3328


# Reflection on results 
Using Pandas to clean and anlayse the financial data.
I checked for missing values and i drop the row with the missing value since the missing value will not have any significant impact on my data.
I then checked for Duplicate but we had no Duplicate in our data_set.
I then moved on to filter my data with pandas for specific condition with a subset of rows which maintain the original structure of the data.
finally, I calculated a summary statistics(Aggregation) to derive insight from the data_set